src/Cmfd.cpp

```diff
@@ -606,311 +601,140 @@ FP_PRECISION Cmfd::computeKeff(int moc_iteration) {
-
-    /* Increment the interations counter */
-    iter++;
-
-    log_printf(DEBUG, "GS iter: %d, res: %f", iter, residual);
-
-    if (residual < conv && iter > 10)
-      break;
-  }
-
-  log_printf(DEBUG, "linear solver iterations: %d", iter);
-}
-
-
-/**
  * @brief Rescale the initial and converged flux arrays.
+ * @details The diffusion problem is a generalized eigenvalue problem and
+ *         therefore the solution is independent of flux level. This method
+ *         rescales the input flux and converged flux to both have an average
+ *         fission source of 1.0 in each group in each cell.
  */
 void Cmfd::rescaleFlux() {
 
-  FP_PRECISION sum_new, sum_old, scale_val;
-
   /* Rescale the new and old flux to have an avg source of 1.0 */
-  matrix_multiplication(_M, _new_flux, _new_source, _num_x*_num_y, 
-      _num_cmfd_groups);
-  sum_new = pairwise_sum(_new_source, _num_x*_num_y*_num_cmfd_groups);
-  scale_val = _num_x*_num_y*_num_cmfd_groups / sum_new;
-  vector_scale(_new_flux, scale_val, _num_x*_num_y*_num_cmfd_groups);
-  matrix_multiplication(_M, _old_flux, _old_source, _num_x*_num_y, 
-          _num_cmfd_groups);
-  sum_old = pairwise_sum(_old_source, _num_x*_num_y*_num_cmfd_groups);
-  scale_val = _num_x*_num_y*_num_cmfd_groups / sum_old;
-  vector_scale(_old_flux, scale_val, _num_x*_num_y*_num_cmfd_groups);
+  matrixMultiplication(_M, _new_flux, _new_source);
+  matrixMultiplication(_M, _old_flux, _old_source);
+
+  _new_flux->scaleByValue(_num_x*_num_y*_num_cmfd_groups /
+                          _new_source->getSum());
+  _old_flux->scaleByValue(_num_x*_num_y*_num_cmfd_groups /
+                          _old_source->getSum());
 }
 
 
-/** @brief Fill in the values in the A matrix, M matrix, and old
- *        scalar flux vector.
+/**
+ * @brief Construct the loss + streaming matrix (A) and the fission gain
+ *         matrix (M) in preparation for solving the eigenvalue problem.
+ * @details This method loops over all mesh cells and energy groups and
+ *         accumulates the iteraction and streaming terms into their
+ *         approipriate positions in the loss + streaming matrix and
+ *         fission gain matrix.
  */
 void Cmfd::constructMatrices() {
 
   log_printf(INFO,"Constructing matrices...");
     
   FP_PRECISION value, volume;
-  int cell, row;
+  int cell_id;
   Material* material;
   
   /* Zero _A and _M matrices */
-  matrix_zero(_M, _num_cmfd_groups*_num_cmfd_groups, _num_x*_num_y);
-  matrix_zero(_A, _num_cmfd_groups*(_num_cmfd_groups+4), _num_x*_num_y);
+  _A->clear();
+  _M->clear();
   
   /* Loop over cells */
-  #pragma omp parallel for private(value, volume, cell, row, material)
+  #pragma omp parallel for private(value, volume, cell_id, material)
   for (int y = 0; y < _num_y; y++) {
     for (int x = 0; x < _num_x; x++) {
 
-      cell = y*_num_x + x;
-      material = _materials[cell];
-      volume = _volumes[cell];
+      cell_id = y*_num_x + x;
```

Comment:
Okay, I'll change in my PR; I will note that this isn't really consistent throughout OpenMOC.